skymp5-server/cpp/server_guest_lib/PieScript.cpp

```diff
@@ -119,15 +256,60 @@ uint32_t PieScript::GetSlotItem(int weaponChance, int armoryChacne,
 {
   std::pair<LootboxItemType, Tier> typeAndTier = AcknowledgeTypeAndTier(
     weaponChance, armoryChacne, consumableChance, nothingChance);
-  int item = GenerateRandomNumber(
-    0, lootTable[typeAndTier.first][typeAndTier.second].size() - 1);
-  return lootTable[typeAndTier.first][typeAndTier.second][item];
+  if (typeAndTier.first != LootboxItemType::Nothing) {
+    int item = GenerateRandomNumber(
+      0, lootTable[typeAndTier.first][typeAndTier.second].size() - 1);
```

Comment:
Let's check that `lootTable[typeAndTier.first][typeAndTier.second].size()` is not zero just in case. And throw if it is. We should not crash server if this suddenly became the case. So let's add check here or in `GenerateRandomNumber` (that L <= R)